src/modules/rlm_eap/types/rlm_eap_ttls/ttls.c

```diff
@@ -629,32 +657,26 @@ fr_radius_packet_code_t eap_ttls_process(request_t *request, eap_session_t *eap_
 	if (data_len == 0) {
 		if (t->authenticated) {
 			RDEBUG2("Got ACK, and the user was already authenticated");
-			code = FR_RADIUS_CODE_ACCESS_ACCEPT;
-			goto finish;
+			return eap_ttls_success(&request->rcode, request, eap_session);
 		} /* else no session, no data, die. */
 
 		/*
 		 *	FIXME: Call SSL_get_error() to see what went
 		 *	wrong.
 		 */
 		RDEBUG2("SSL_read Error");
-		code = FR_RADIUS_CODE_ACCESS_REJECT;
-		goto finish;
+		return UNLANG_ACTION_FAIL;
 	}
 
-	if (!diameter_verify(request, data, data_len)) {
-		code = FR_RADIUS_CODE_ACCESS_REJECT;
-		goto finish;
-	}
+	if (!diameter_verify(request, data, data_len)) return UNLANG_ACTION_FAIL;
```

Comment:
Nit: do we eventually want to add a `src/protocols/diameter`, and move the verify / encode / decode there?

Not for this  commit, but something to think about.

Reply:
Yeah I was thinking exactly the same.  Probably not too much work to shift the code over and fixup up the function signature.

Reply:
...then we could stick someone on starting to write tests, and making the code more generic.